src/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.cs

```diff
@@ -0,0 +1,233 @@
+            using (ZipArchive archive = new ZipArchive(await StreamHelpers.CreateTempCopyStream(ZipTest.zfile("small.zip")), ZipArchiveMode.Update))
+            {
+                ZipArchiveEntry entry = archive.Entries[0];
+                String contents1, contents2;
+                using (StreamReader s = new StreamReader(entry.Open()))
+                {
+                    contents1 = s.ReadToEnd();
+                }
+                using (StreamReader s = new StreamReader(entry.Open()))
+                {
+                    contents2 = s.ReadToEnd();
+                }
+                Assert.Equal(contents1, contents2);
+            }
+        }
+
+        [Fact]
+        public static async Task UpdateCreate()
+        {
+            await testFolder("normal");
+            await testFolder("empty");
+            await testFolder("unicode");
+        }
+
+        public static async Task testFolder(String s)
+        {
+            var zs = new LocalMemoryStream();
+            await ZipTest.CreateFromDir(ZipTest.zfolder(s), zs, ZipArchiveMode.Update);
+            ZipTest.IsZipSameAsDir(zs.Clone(), ZipTest.zfolder(s), ZipArchiveMode.Read, false, false);
+        }
+
+
+        [Fact]
+        public static void UpdateEmptyEntry()
+        {
+            ZipTest.EmptyEntryTest(ZipArchiveMode.Update);
+        }
+
+        [Fact]
+        public static async Task UpdateModifications()
+        {
+            //delete and move
+            var testArchive = await StreamHelpers.CreateTempCopyStream(ZipTest.zfile("normal.zip"));
+
+            using (ZipArchive archive = new ZipArchive(testArchive, ZipArchiveMode.Update, true))
+            {
+                ZipArchiveEntry toBeDeleted = archive.GetEntry("binary.wmv");
+                toBeDeleted.Delete();
+                toBeDeleted.Delete(); //delete twice should be okay
+                ZipArchiveEntry moved = archive.CreateEntry("notempty/secondnewname.txt");
+                ZipArchiveEntry orig = archive.GetEntry("notempty/second.txt");
+                using (Stream origMoved = orig.Open(), movedStream = moved.Open())
+                {
+                    origMoved.CopyTo(movedStream);
+                }
+                moved.LastWriteTime = orig.LastWriteTime;
+                orig.Delete();
+            }
+
+            ZipTest.IsZipSameAsDir(testArchive, ZipTest.zmodified("deleteMove"), ZipArchiveMode.Read, false, false);
+            
+            //append
+            testArchive = await StreamHelpers.CreateTempCopyStream(ZipTest.zfile("normal.zip"));
+
+            using (ZipArchive archive = new ZipArchive(testArchive, ZipArchiveMode.Update, true))
+            {
+                ZipArchiveEntry e = archive.GetEntry("first.txt");
+
+                using (StreamWriter s = new StreamWriter(e.Open()))
+                {
+                    s.BaseStream.Seek(0, SeekOrigin.End);
+
+                    s.Write("\r\n\r\nThe answer my friend, is blowin' in the wind.");
```

Comment:
In this case, it looks like it should stay as-is, as it appears to be generating a zip file that's then being compared against an existing asset, which presumably was generated using the same input. We can fix it layer if need be when actually testing on another OS.